### DISCUSSION TITLE
minio endpoint for s3adapter

### DISCUSSION THREAD

**Discussion by ChristoVV** [payloadcms/payload#4866] [Q&A]
Hello guys could you guys help me pls? I feel like I'm doing something wrong. I am creating my s3 adapter from plugin-cloud-storage:
```
const storageAdapter = s3Adapter({
  config: {
    endpoint: process.env.S3_ENDPOINT,
    region: 'us-east-1',
    credentials: {
      accessKeyId: process.env.S3_ACCESS_KEY,
      secretAccessKey: process.env.S3_SECRET_KEY,
    },
  },
  bucket: process.env.S3_BUCKET_NAME,
});
```

And use it in payload config in plugins:

```
plugins: [
    payloadCloud(),
    cloudStorage({
      collections: {
        media: {
          adapter: storageAdapter,
        },
      },
    }),
  ],
```
but i get error in console when i try to upload some files in media:
```
[14:23:13] ERROR (payload): getaddrinfo ENOTFOUND (my address)     
    err: {
      "type": "Error",
      "message": "getaddrinfo ENOTFOUND (my address)",
      "stack":
          Error: getaddrinfo ENOTFOUND (my address)
              at GetAddrInfoReqWrap.onlookupall [as oncomplete] (node:dns:118:26)
      "errno": -3008,
      "code": "ENOTFOUND",
      "syscall": "getaddrinfo",
      "hostname": "(my address)",
      "$metadata": {
        "attempts": 1,
        "totalRetryDelay": 0
      }
    }
```
Instead of (my address), there is a link to my bucket everywhere which I specifically replaced with (my address). Please tell me, did I forget to add some option to s3adapter or am I just specifying the endpoint incorrectly? if I specify the endpoint incorrectly, please give me an example of the correct endpoint for a minIO. I read the documentation for plugin-cloud-storage and saw that for S3-Compatible Storage, such as minIO you just need to add an endpoint, but it doesn't work for me.

**Comment by ChristoVV** [comment]
I solved the problem, I had to add
`forcePathStyle: true`
sorry, I don't know how to delete the discussion, if you can, delete it yourself, or leave it, maybe it will be useful to someone

**Comment by iFoxtrot33** [comment]
Thank you! This helped me! Do not delete the discussion =)

**Comment by ron-tayler** [comment]
Спасибо , помогло

**Comment by magicspon** [comment]
FYI. This also fixes rustfc.
